File: app/src/main/java/com/example/kaliumapp/data/database/entities/CalorieExpenditure.kt
package com.example.kaliumapp.data.database.entities

import androidx.room.*

@Entity(
    tableName = "calorie_expenditure",
    foreignKeys = [ForeignKey(
        entity = User::class,
        parentColumns = ["id"],
        childColumns = ["userId"],
        onDelete = ForeignKey.CASCADE
    )],
    indices = [Index(value = ["userId"])]
)
data class CalorieExpenditure(
    @PrimaryKey(autoGenerate = true)
    val id: Int = 0,
    val userId: Int,
    val calories: Float,
    val date: String,
    val timestamp: Long = System.currentTimeMillis()
)

File: models/calorie_expenditure.py
from sqlalchemy import Column, Integer, Float, String, ForeignKey
from database.database import Base

class CalorieExpenditure(Base):
    __tablename__ = "calorie_expenditure"
    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), index=True)
    calories = Column(Float, nullable=False)
    date = Column(String, nullable=False)